### DISCUSSION TITLE
More detailed error messages for failed httpfs fetches

### DISCUSSION THREAD

**Discussion by rkennedy-mode** [duckdb/duckdb#9865] [Ideas]
I was testing something locally (via the JDBC driver) with read_csv loading data from an HTTP endpoint. The loading was failing with the following error:

```
java.sql.SQLException: IO Error: Unable to connect to URL "http://localhost:62737/data.csv": 404 (Not Found)
```

If I requested that URL via cURL it fetched just fine. It wasn't until a bit of tinkering with the HTTP server that I realized DuckDB was issuing an HTTP HEAD request, not an HTTP GET request (presumably some sort of pre-flight request).

The feature request is for the error message to include more details of the failed request, namely the HTTP method being used. That would have saved me a solid 15-30 minutes of head scratching. 😄

**Comment by carlopi** [comment]
Infrastructure is already there for collecting this information, and this was exposed to users up to DuckDB v0.9.0.

It has been temporarily disabled since there was a bug connected to this, but it should be possible (and desirable) to restore more helpful behaviour.

Thanks for the report / feedback

**Reply by rkennedy-mode** [reply]
> this was exposed to users up to DuckDB v0.9.0

Do you have a pointer to where this is exposed?

**Comment by carlopi** [comment]
PR removing it was here: https://github.com/duckdb/duckdb/pull/9351

**Comment by carlopi** [comment]
Proper support for error messages carrying over more informations has been reintegrated thanks to a recent PR by @Mytherin. This will be part of the upcoming v0.10.0
